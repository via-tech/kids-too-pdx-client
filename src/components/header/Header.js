import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from './Header.css';
import logo from '../../../public/assets/logo.png';

function Header({ token, handleSignOut }) {
  return (
    <>
        <section className={styles.container}>
          <header className={styles.Header}>
            <section className={styles.logo}>
              <img src={logo} />
            </section>
            {!token && <Link to="/login">Sign In</Link>}
            {!token && <Link to="/signup">Sign Up</Link>}
            {token && <button onClick={handleSignOut}>Sign Out</button>}
            <nav className={styles.nav}>
              <Link to="/">Home</Link>
              <Link to='/submit-event'>Submit Event</Link>
              <Link to="/contact">Contact</Link>
              <Link to='/resources'>Resources</Link>
            </nav>
          </header>
        </section>
      </>
  );
}

Header.propTypes = {
  token: PropTypes.string,
  handleSignOut: PropTypes.func
};

export default Header;
