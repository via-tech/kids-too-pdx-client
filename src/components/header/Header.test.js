import React from 'react';
import renderer from 'react-test-renderer';
import Header from './Header';
import { MemoryRouter } from 'react-router-dom';

describe('Header', () => {
  it('snapshot Header signed out', () => {
    const props = {
      token: '',
      handleSignOut: jest.fn()
    };

    const tree = renderer.create(
      <MemoryRouter>
        <Header {...props} />
      </MemoryRouter>
    );

    expect(tree).toMatchSnapshot();
  });

  it('snapshots Header signed in', () => {
    const props = {
      token: '1234',
      handleSignOut: jest.fn()
    };

    const tree = renderer.create(
      <MemoryRouter>
        <Header {...props} />
      </MemoryRouter>
    );

    expect(tree).toMatchSnapshot();
  });
});
