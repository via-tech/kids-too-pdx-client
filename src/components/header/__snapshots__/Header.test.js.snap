// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot Header signed out 1`] = `
<section
  className="container"
>
  <header
    className="Header"
  >
    <section
      className="logo"
    >
      <img
        src="/path/to/image.png"
      />
    </section>
    <a
      href="/login"
      onClick={[Function]}
    >
      Sign In
    </a>
    <a
      href="/signup"
      onClick={[Function]}
    >
      Sign Up
    </a>
    
    <nav
      className="nav"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        href="/submit-event"
        onClick={[Function]}
      >
        Submit Event
      </a>
      <a
        href="/contact"
        onClick={[Function]}
      >
        Contact
      </a>
      <a
        href="/resources"
        onClick={[Function]}
      >
        Resources
      </a>
    </nav>
  </header>
</section>
`;

exports[`Header snapshots Header signed in 1`] = `
<section
  className="container"
>
  <header
    className="Header"
  >
    <section
      className="logo"
    >
      <img
        src="/path/to/image.png"
      />
    </section>
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        onClick={[MockFunction]}
      >
        Sign Out
      </button>
    </a>
    <nav
      className="nav"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        href="/submit-event"
        onClick={[Function]}
      >
        Submit Event
      </a>
      <a
        href="/contact"
        onClick={[Function]}
      >
        Contact
      </a>
      <a
        href="/resources"
        onClick={[Function]}
      >
        Resources
      </a>
    </nav>
  </header>
</section>
`;
